src/coreComponents/mesh/generators/WellBlockABC.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_WELLBLOCKABC_HPP
+#define GEOSX_WELLBLOCKABC_HPP
+
+#include "dataRepository/Group.hpp"
+#include "mesh/ElementType.hpp"
+#include "common/DataTypes.hpp"
+#include "dataRepository/Group.hpp"
+
+#include <vector>
+
+namespace geosx
+{
+
+/**
+ * Abstract base class defining the information provided by any well block implementation.
+ *
+ * It's noteworthy that the WellblockABC is immutable oriented.
+ * The derived implementations need to have the modification/creation capabilities.
+ */
+class WellBlockABC
```

Comment:
@francoishamon what do you think of the interface of `WellBlockABC`? Does it make sense to you?